modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/MajorityPeerSelect.scala

```diff
@@ -50,28 +60,32 @@ object MajorityPeerSelect {
 
     def filterPeerList(peers: NonEmptyList[Peer]): F[NonEmptyList[Peer]] =
       peers
-        .traverse(snapshotClient.getLatestOrdinal(_))
-        .map {
-          _.groupBy(identity).maxBy { case (_, ordinals) => ordinals.size }
+        .parTraverseN(maxConcurrentPeerInquiries)(snapshotClient.getLatestOrdinal(_))
+        .map(_.groupBy(identity).maxBy { case (_, ordinals) => ordinals.size })
+        .map { case (majorityOrdinal, _) => majorityOrdinal }
+        .flatMap { majorityOrdinal =>
+          peers
+            .parTraverseN(maxConcurrentPeerInquiries)(peer => getPeerHash(peer, majorityOrdinal).attempt)
         }
-        .flatMap {
-          case (majorityOrdinal, _) =>
-            peers.traverse(snapshotClient.get(majorityOrdinal).run(_).flatMap(_.toHashed.map(_.hash)))
+        .flatMap { maybePeerSnapshots =>
+          MonadThrow[F].fromOption(
+            maybePeerSnapshots.collect { case Right(peerSnapshot) => peerSnapshot }.toNel,
```

Comment:
I think that `maybePeerSnapshots.sequence.toNel` would do the same (or flatSequence)

Reply:
also, `maybe` suggests it's an optional type, but collect uses `either`. so is it an Option or Either?

Reply:
`maybePeerSnapshots` renamed to `peerSnapshotHashes` and `maybePeerSnapshots.collect { case Right(peerSnapshot) => peerSnapshot }.toNel` changed to `peerSnapshotHashes.sequence.toOption`.